### ISSUE TITLE
what about 4k resulation fixes

### ISSUE THREAD

**Issue by baadshah0786** [yuliskov/SmartTube#3698]
Bro plz do something for fix 4k videos resulation

**Comment by Flame-Rider** [comment]
@https://github.com/yuliskov Dev needs to fix this. it is pure laziness every other app had day 1 fixes for 4k issues

**Comment by Flame-Rider** [comment]
> @Flame-Rider how much are you paying monthly to use the app?
> 
> You know that the developer does this a an hobby on his spare time, also he's from Ukraine sometimes not even electricity is working due to some complications.
> 
> So stop being lazy and fix the issue yourself and create a PR

How about you do it yourself instead of criticizing me. Every other app has a fix day 1 yet the "fix" for this does not even have hdr or any other things at this point that it used to yet every other app does day 1 and you actually bother to get some intelligence and know what you are talking about before posting and knowing what your talking about the whole point of the app is not to pay for something that is not supposed to be paid for to begin with so you can shove it

**Comment by yuliskov** [comment]
Guys, sorry for the delay with the 4K fix. I'm working on it.

**Comment by PowerWiesel** [comment]
@Flame-Rider  screw you! holy shit how can anyone be this entitled?

@yuliskov I'm so sorry you have to deal with those wet wipes. Thanks you s much for your amazing work.

**Comment by thestuartcarson** [comment]
> Guys, sorry for the delay with the 4K fix. I'm working on it.

Only one person here in this thread should apologise and it’s not you. You are doing phenomenal work, and we thank you for it.

**Comment by PowerWiesel** [comment]
> > Guys, sorry for the delay with the 4K fix. I'm working on it.
> 
> Only one person here in this thread should apologise and it’s not you. You are doing phenomenal work, and we thank you for it.

right? it makes my blood boil when I read what those idiots post.

Thanks again @yuliskov

**Comment by thestuartcarson** [comment]
> > > Guys, sorry for the delay with the 4K fix. I'm working on it.
> > 
> > 
> > Only one person here in this thread should apologise and it’s not you. You are doing phenomenal work, and we thank you for it.
> 
> right? it makes my blood boil when I read what those idiots post.
> 
> Thanks again @yuliskov

When he missed your point completely about how this amazing guy does all this for free, that’s what pissed me off most.

**Comment by rodalpho** [comment]
@yuliskov  please don't apologize! Thanks for all your work and stay safe.

**Comment by sakiskom** [comment]
> Guys, sorry for the delay with the 4K fix. I'm working on it.
My friend, i want to tell you a BIG thank you for the great app!

Thank you!

**Comment by andreferchau** [comment]
I am so thankful for your app! 

"I" was guilty of not even supporting you. 
When I realized how quickly you came out with a fix I just had to sign up a subscription and help financially help you a little bit. 

Thank you thank you so much
I will never complain about your work

**Comment by a87720** [comment]
@yuliskov 真诚感谢你所做出的一切！

**Comment by alcatraz-zz** [comment]
I have a question. You guys refer to this as 4k being unavailable. For me even at 1080p I'm limited. Limited only to AVC streams. VP9 and AV1 are unavailable. Is that what you're getting?

Also, only v22.38 plays videos. 22.39-22.41 all give me 403 errors. Is that the same with you? Thanks! I'm having some issues with my tv box and I'm not sure it's box or app related. Thanks.

**Comment by nuson999** [comment]
> I have a question. You guys refer to this as 4k being unavailable. For me even at 1080p I'm limited. Limited only to AVC streams. VP9 and AV1 are unavailable. Is that what you're getting?
> 
> Also, only v22.38 plays videos. 22.39-22.41 all give me 403 errors. Is that the same with you? Thanks! I'm having some issues with my tv box and I'm not sure it's box or app related. Thanks.

I'm also experiencing intermittent 403 errors when playing videos in 1080p
v22.41

**Comment by Grumpy-Geoff** [comment]
> 22.39-22.41 all give me 403 errors

What I found is if you leave the "403" error displaying and don't do anything then after a few seconds the video will start to play.

**Comment by josejoa** [comment]
> Guys, sorry for the delay with the 4K fix. I'm working on it.

I will donate for Ukraine and SmartTube, take care!

**Comment by iDoNtCaRe-69** [comment]
> @yuliskov Dev needs to fix this. it is pure laziness every other app had day 1 fixes for 4k issues

If you don't know already, YouTube ATV API was attacked by google recently. An temporary fix was implemented to fix that, which resulted in 1080P video quality. 

Cause: #3626 
Symptoms: Anything but Live Stream will reduse to play.

If you have a fix, Please share it with us. Otherwise don't like you won anybody's ass. You can use any older build that didn't have this patch implemented, and enjoy your 4K HDR Playback you moron 🤡

**Comment by alcatraz-zz** [comment]
> You can use any older build that didn't have this patch implemented, and enjoy your 4K HDR Playback 

Do you mean that there's an older version of SmartTube that still works in 4k? Like one that uses an older API.

**Comment by iDoNtCaRe-69** [comment]
@alcatraz-zz obviously

**Comment by alcatraz-zz** [comment]
> @alcatraz-zz obviously

Cool!

Which version?

**Comment by Flame-Rider** [comment]
> > @yuliskov Dev needs to fix this. it is pure laziness every other app had day 1 fixes for 4k issues
> 
> If you don't know already, YouTube ATV API was attacked by google recently. An temporary fix was implemented to fix that, which resulted in 1080P video quality.
> 
> Cause: #3626 Symptoms: Anything but Live Stream will refuse to play.
> 
> If you have a fix, Please share it with us. Otherwise don't act like you own anybody's ass. You can use any older build that didn't have this patch implemented, and enjoy your 4K HDR Playback you moron 🤡

I am not acting like i own anyone's ass i don't give a fuck why, every other app has a fix for this out day 1. People need to start thinking for themselves and actually show some intelligence for once and expecting quality is not asking much at all whether something is free or paid does not matter if you make something it should be made of quality otherwise you just give devs a bad name i couldn't give a fuck what you personally believe it is not my problem i am not the one who started shit all of you idiots started spouting bullshit so i spoke out. You can eat my ass

**Comment by iDoNtCaRe-69** [comment]
> > > @yuliskov Dev needs to fix this. it is pure laziness every other app had day 1 fixes for 4k issues
> > 
> > 
> > If you don't know already, YouTube ATV API was attacked by google recently. An temporary fix was implemented to fix that, which resulted in 1080P video quality.
> > Cause: #3626 Symptoms: Anything but Live Stream will refuse to play.
> > If you have a fix, Please share it with us. Otherwise don't act like you own anybody's ass. You can use any older build that didn't have this patch implemented, and enjoy your 4K HDR Playback you moron 🤡
> 
> I am not acting like i own anyone's ass i don't give a fuck why, every other app has a fix for this out day 1. People need to start thinking for themselves and actually show some intelligence for once and expecting quality is not asking much at all whether something is free or paid does not matter if you make something it should be made of quality otherwise you just give devs a bad name i couldn't give a fuck what you personally believe it is not my problem i am not the one who started shit all of you idiots started spouting bullshit so i spoke out. You can eat my ass

Go on, use THAT APP which already has the fix. Who's stopping you from doing that? 🤡

**Comment by rodalpho** [comment]
Heh in my experience once analingus enters the conversation there's not much point in responding further.

**Comment by Flame-Rider** [comment]
> @Flame-Rider screw you! holy shit how can anyone be this entitled?
> 
> 

The only idiots here are you baboons that can't make coherent thoughts letting people walk all over you and think sloppiness and meager work is okay expecting quality is not entitlement. Do you hear how stupid you sound

**Comment by alcatraz-zz** [comment]
> > @Flame-Rider screw you! holy shit how can anyone be this entitled?
> 
> The only idiots here are you baboons that can't make coherent thoughts letting people walk all over you and think sloppiness and meager work is okay expecting quality is not entitlement. Do you hear how stupid you sound

Please be nice to the dev. 

Why not try the official Youtube for Android "app" in the meantime. Sure the ads are annoying but you get full 4k and all the fancy codecs like AV1.

If Smarttube needs more time maybe another third party app will emerge. That'd be cool. More choices for us!

Does anybody know why Youtube Revanced (ext) only does AVC up to 1080p streams on my TV box? Youtube ATV does VP9/AV1 and Smarttube used to do it.

**Comment by josejoa** [comment]
> > > @Flame-Rider screw you! holy shit how can anyone be this entitled?
> > 
> > 
> > The only idiots here are you baboons that can't make coherent thoughts letting people walk all over you and think sloppiness and meager work is okay expecting quality is not entitlement. Do you hear how stupid you sound
> 
> Please be nice to the dev.
> 
> Why not try the official YoutubeTV "app" in the meantime? Sure the ads are annoying but you get full 4k and all the fancy codecs like AV1.
> 
> If Smarttube needs more time maybe another third party app will emerge. That'd be cool. More choices for us!
> 
> Does anybody know why Youtube Revanced (ext) only does AVC up to 1080p streams on my TV box? YoutubeTV does VP9/AV1 and Smarttube used to do it.

None app like this can't reach 4K at the moment: Revanced, NewPipe, Tubular...

**Comment by alcatraz-zz** [comment]
> None app like this can't reach 4K at the moment: Revanced, NewPipe, Tubular...

But the same app does 4k, VP9 on my phone right now. On the tv box it's up to 1080p AVC. I don't understand why.

**Comment by iDoNtCaRe-69** [comment]
> > None app like this can't reach 4K at the moment: Revanced, NewPipe, Tubular...
> 
> But the same app does 4k, VP9 on my phone right now. On the tv box it's up to 1080p AVC. I don't understand why.

Contact google customer service and ask them to stop targeting android TV users from Ad Blocking stuff. It's their fault to begin with.

**Comment by josejoa** [comment]
> > None app like this can't reach 4K at the moment: Revanced, NewPipe, Tubular...
> 
> But the same app does 4k, VP9 on my phone right now. On the tv box it's up to 1080p AVC. I don't understand why.

Revanced mostly is for phones. In TV, YouTube uses WebView and works different than an app, due to that developers find some problems to overcome the limit of 1080.

**Comment by alcatraz-zz** [comment]
> > > None app like this can't reach 4K at the moment: Revanced, NewPipe, Tubular...
> > 
> > 
> > But the same app does 4k, VP9 on my phone right now. On the tv box it's up to 1080p AVC. I don't understand why.
> 
> Contact google customer service and ask them to stop targeting android TV users from Ad Blocking stuff. It's their fault to begin with.



> > > None app like this can't reach 4K at the moment: Revanced, NewPipe, Tubular...
> > 
> > 
> > But the same app does 4k, VP9 on my phone right now. On the tv box it's up to 1080p AVC. I don't understand why.
> 
> Revanced mostly is for phones. In TV, YouTube uses WebView and works different than an app, the web server can change very fast and suddenly, due to this the developers find some problems to overcome the limit of 1080.

Maybe root can be used to launch the app in phone mode. My box is rooted.

**Comment by alcatraz-zz** [comment]
Hey guys. The new beta 22.43 works great for me. (available when updating within the app)

**Comment by josejoa** [comment]
Hey guys. The new beta 22.43 works great for me. (available when updating within the app)

You are right, beta 22.43 works fine in 4K. We have to thank to this great developer his work [yuliskov](https://github.com/yuliskov)

**Comment by josejoa** [comment]
> Hey guys. The new beta 22.43 works great for me. (available when updating within the app)

You are right, 22.43 beta works fine in 4K. We have to thank this great developer for his work [yuliskov](https://github.com/yuliskov)

**Comment by alcatraz-zz** [comment]
Woops. The fix was short lived. Now I'm back to 1080p avc only on v22.44.

Google must really be waging war right now. 😎

**Comment by josejoa** [comment]
> Woops. The fix was short lived. Now I'm back to 1080p avc only on v22.44.
> 
> Google must really be waging war right now. 😎

No. I have the stable version 22.44 and I can see videos in 4K, I have also a TV Box.

**Comment by alcatraz-zz** [comment]
> > Woops. The fix was short lived. Now I'm back to 1080p avc only on v22.44.
> > Google must really be waging war right now. 😎
> 
> No. I have the stable version 22.44 and I can see videos in 4K, I have also a TV Box.

That's odd. I upgraded to 22.47 this morning and I had 4k resolutions back. Yesterday evening I only had 1080p on 22.44.

**Comment by josejoa** [comment]
> > > Woops. The fix was short lived. Now I'm back to 1080p avc only on v22.44.
> > > Google must really be waging war right now. 😎
> > 
> > 
> > No. I have the stable version 22.44 and I can see videos in 4K, I have also a TV Box.
> 
> That's odd. I upgraded to 22.47 this morning and I had 4k resolutions back. Yesterday evening I only had 1080p on 22.44.

22.47 is beta, as long as I can see 4K on the stable version (22.44), I won't go back to beta.

**Comment by alcatraz-zz** [comment]
> 22.47 is beta, as long as I can see 4K on the stable version (22.44), I won't go back to beta. Unfortunately I find some bugs in the stable version and also in the beta version (22.48 right now), it says "Unknow source...", a string quite long.

It happens to me too but usuallt it just restarts by itself. I have official YoutubeTV ready in case Smarttube stops working. I can recommend it. That way you can always enjoy higher resolutions.

**Comment by josejoa** [comment]
> > 22.47 is beta, as long as I can see 4K on the stable version (22.44), I won't go back to beta. Unfortunately I find some bugs in the stable version and also in the beta version (22.48 right now), it says "Unknow source...", a string quite long.
> 
> It happens to me too but usually it just restarts by itself = not a massive problem. I have official YoutubeTV ready in case Smarttube stops working. I can recommend it. That way you can always enjoy higher resolutions.

You can also use a good browser for TV, I recommend Zeus TV Browser, It blocks all ads and you can enjoy videos in 4K, even in full screen.

**Comment by alcatraz-zz** [comment]
> > > 22.47 is beta, as long as I can see 4K on the stable version (22.44), I won't go back to beta. Unfortunately I find some bugs in the stable version and also in the beta version (22.48 right now), it says "Unknow source...", a string quite long.
> > 
> > 
> > It happens to me too but usually it just restarts by itself = not a massive problem. I have official YoutubeTV ready in case Smarttube stops working. I can recommend it. That way you can always enjoy higher resolutions.
> 
> You can also use a good browser for TV, I recommend Zeus TV Browser, It blocks all ads and you can enjoy videos in 4K, even in full screen.

Interesting. I wonder if it can do AV1 hardware decoding....

**Comment by josejoa** [comment]
> > > > 22.47 is beta, as long as I can see 4K on the stable version (22.44), I won't go back to beta. Unfortunately I find some bugs in the stable version and also in the beta version (22.48 right now), it says "Unknow source...", a string quite long.
> > > 
> > > 
> > > It happens to me too but usually it just restarts by itself = not a massive problem. I have official YoutubeTV ready in case Smarttube stops working. I can recommend it. That way you can always enjoy higher resolutions.
> > 
> > 
> > You can also use a good browser for TV, I recommend Zeus TV Browser, It blocks all ads and you can enjoy videos in 4K, even in full screen.
> 
> Interesting. I wonder if it can do AV1 hardware decoding....

I don't know if it has AV1, I don't know what uses YouTube, but you can enjoy videos in 4K, like in the desktop version. You can select in settings of the player on the link, then quality, and choose 2160p60, this is 4K, [https://www.youtube.com/watch?v=wHS7zpvgNvA](url)
